editor/code_editor.cpp

```diff
@@ -1608,6 +1618,94 @@ void CodeTextEditor::duplicate_selection() {
 	text_editor->queue_redraw();
 }
 
+void CodeTextEditor::duplicate_lines(bool p_below) {
+	text_editor->begin_complex_operation();
+
+	Vector<int> caret_edit_order = text_editor->get_caret_index_edit_order();
+	for (const int &caret_index : caret_edit_order) {
+		// The text that will be inserted. All lines in one string
+		String insert_text;
+
+		// The new line position of the caret after the operation
+		int new_caret_line = text_editor->get_caret_line(caret_index);
+		// The new column position of the caret after the operation
+		int new_caret_column = text_editor->get_caret_column(caret_index);
+		// The caret positions of the selection. Stays -1 if there is no selection
+		int select_from_line = -1;
+		int select_to_line = -1;
+		int select_from_column = -1;
+		int select_to_column = -1;
```

Comment:
I tried to remove them, but when the caret moves, the selection gets unset, which means that I need to store them before changing the caret. And I think that I should transfer the selection to the duplication since it means that you can perform the same duplication multiple times in a row. Which is something I use very often.